### ISSUE TITLE
Import number of events from RSD

### ISSUE THREAD

**Issue by Darkless012** [gis4dis/poster#2]
<a href="https://github.com/jirik"><img src="https://avatars0.githubusercontent.com/u/2271850?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [jirik](https://github.com/jirik)**
_Friday Oct 13, 2017 at 11:59 GMT # Sample: Friday Sep 13, 2013 at 22:58 GMT_
_Originally opened as https://github.com/gis4dis/poster_obsolete/issues/6_

----

We are interested in number of new events that appeared in RSD logs
- per category
- per hour
- per municipality (Brno + Brno-venkov + D1 municipalities)

Initial steps
- fork this repo
- create branch for this issue
- create custom commands for exploring RSD data in https://github.com/gis4dis/poster/tree/master/src/apps/processing/rsd
  - inspire in [ALA](https://github.com/gis4dis/poster/tree/master/src/apps/processing/ala/management/commands) or [O2](https://github.com/gis4dis/poster/tree/master/src/apps/processing/o2/management/commands) commands
  - use [models of provider logs](https://github.com/gis4dis/poster/blob/master/src/apps/importing/models.py) for reading logs from database
  - one of important questions we need to answer is: Is any RSD event related always to single municipality, or may it be related to multiple municipalities?

**Comment by Darkless012** [comment]
<a href="https://github.com/etomanon"><img src="https://avatars2.githubusercontent.com/u/17390813?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [etomanon](https://github.com/etomanon)**
_Friday Feb 02, 2018 at 12:30 GMT # Sample: Friday Sep 13, 2013 at 22:58 GMT_

----

RSD event can be related to multiple municipalities. For example, 'D1, mezi km 150 a 0, nárazový vítr, Od 18.01.2018 18:31 Do 19.01.2018 06:00' event log has 57 TownCodes.

**Comment by Darkless012** [comment]
<a href="https://github.com/jirik"><img src="https://avatars0.githubusercontent.com/u/2271850?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jirik](https://github.com/jirik)**
_Monday Feb 05, 2018 at 10:12 GMT # Sample: Friday Sep 13, 2013 at 22:58 GMT_

----

Thanks for this information. We'll discuss what it means for models on Wednesday. Suggestions welcome.

**Comment by Darkless012** [comment]
<a href="https://github.com/etomanon"><img src="https://avatars2.githubusercontent.com/u/17390813?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [etomanon](https://github.com/etomanon)**
_Thursday Feb 08, 2018 at 20:17 GMT # Sample: Friday Sep 13, 2013 at 22:58 GMT_

----

Attribute for category (TXUCL) of an event is optional but attribute can have multiple categories.
Here are the logs for rsd_import command - 3 days (without D1)
[logs.txt](https://github.com/gis4dis/poster_obsolete/files/1708632/logs.txt)

**Comment by Darkless012** [comment]
<a href="https://github.com/etomanon"><img src="https://avatars2.githubusercontent.com/u/17390813?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [etomanon](https://github.com/etomanon)**
_Friday Feb 09, 2018 at 16:32 GMT # Sample: Friday Sep 13, 2013 at 22:58 GMT_

----

Here are logs for new categories command (including D1)
Category attribute is optional according to RSD documentation but I have not found an event log without TXUCL tag.
[categories.txt](https://github.com/gis4dis/poster_obsolete/files/1711673/categories.txt)

**Comment by jirik** [comment]
@etomanon Thanks for this research. Let's continue with #9 and #10.

It would be great also to prepare pull request with command(s) you already implemented for two logs mentioned above.

**Comment by etomanon** [comment]
Create EventObservations from ProviderLogs:
Link to shapefiles: https://drive.google.com/file/d/1NTj1ALi58v1Re2NkK3flBgxZ7s-jWdGr/view?usp=sharing
1, Run 2x (municipalities + districts) import_towns command 
( ./dcmanage.sh --path 'var/data/obce_4326.shp' )
2, Run import_observed to create metadata
3, Run import_categories
4, Run import_extents
5, Run import_events
6, Run show_events e.g.: ./dcmanage.sh show_events --geometry 'POINT(16.597252 49.157379)' --categories '201, 211' --time_range_start "1 1 2018 12:00" --time_range_end "28 1 2018 20:00" --buffer 10000

**Comment by jirik** [comment]
```
flynn run python manage.py import_towns --path shapefiles/rsd/obce_4326.shp
flynn run python manage.py import_towns --path shapefiles/rsd/momc_4326.shp
flynn run python manage.py import_observed
flynn run python manage.py import_categories
```

**Comment by jirik** [comment]
```bash
flynn run python manage.py import_extents
# ~ 4 min

flynn run python manage.py import_events
# ~ 35 min

flynn run python manage.py show_events --geometry 'POINT(16.597252 49.157379)' --categories '201,211' --time_range_start "1 1 2018 12:00" --time_range_end "28 1 2018 20:00" --buffer 10000
```

**Comment by jirik** [comment]
Closed by #15
